src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -616,26 +612,19 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
   def parseMethod(): Unit = {
     val jflags = readMethodFlags()
     val sflags = jflags.toScalaFlags
-    if (jflags.isPrivate) {
-      val isConstructor = pool.getName(u2()).value == "<init>" // opt avoid interning a Name for private methods we're about to discard
-      if (isConstructor)
-        sawPrivateConstructor = true
-      in.skip(2); skipAttributes()
+    if (jflags.isPrivate || ((sflags & PRIVATE) != 0L)) {
```

Comment:
Not sure if these tests are meaningfully distinct.

Reply:
They are the same 👍

Reply:
Thanks LKAS @lrytz